http2 support doretry when connection reset by peer or broken pipe

### Describe the bug
```
6543 2021-06-15 15:47:11,894 [ERROR] [network] [write directly] Error on write. Connection = 9, Remote Address = 127.0.0.1:8081, err = writev tcp 127.0.0.1:36479->127.0.0.1:80        81: writev: connection reset by peer, conn = 0xc0002ee480
6546 2021-06-15 15:47:11,894 [ERROR] [7,-] http2 client endStream error = writev tcp 127.0.0.1:36479->127.0.0.1:8081: writev: connection reset by peer
6906 2021-06-15 15:47:11,895 [DEBUG] http2 client recevice goaway lastStremID = 199
6954 2021-06-15 15:47:11,895 [DEBUG] [network] [read loop] Error on read. Connection = 9, Local Address = 127.0.0.1:36479, Remote Address = 127.0.0.1:8081, err = EOF
```
当在很高并发的时候，在APP服务端发送了goaway帧然后关闭socket之后，mosn在读取到goaway帧和Close之前，如上日志，有1ms的间隔，会一直使用这条关闭的连接，导致出现`writev: connection reset by peer`的报错，针对这种临界场景，可认为是连接层面的失败，需要支持重试，减少这种失败。

### Expected behavior

### Actual behavior

### Steps to reproduce

### Minimal yet complete reproducer code (or GitHub URL to code)

### Environment

- MOSN Version 0.23.0

--- a/pkg/stream/http2/stream.go
+++ b/pkg/stream/http2/stream.go
@@ -1021,9 +1021,14 @@ reset:
 	log.Proxy.Errorf(s.ctx, "http2 client endStream error = %v", err)
 	if err == types.ErrConnectionHasClosed || err == errClosedClientConn {
 		s.ResetStream(types.StreamConnectionFailed)
-	} else {
-		s.ResetStream(types.StreamLocalReset)
+		return
 	}
+	// fix: https://github.com/mosn/mosn/issues/1672
+	if strings.Contains(err.Error(), "broken pipe") || strings.Contains(err.Error(), "connection reset by peer") {
+		s.ResetStream(types.StreamConnectionFailed)
+		return
+	}
+	s.ResetStream(types.StreamLocalReset)
 }
 
 func (s *clientStream) GetStream() types.Stream {

